spec/json_spike/json_validator_spec.rb

```diff
@@ -0,0 +1,93 @@
+require 'rspec'
+
+describe 'JsonValidator' do
+  let(:payload) do
+    {
+      applicant: {
+        date_of_birth: dob,
+        involvement_type: involvement_type,
+        has_partner_opponent: opponent,
+        receives_qualifying_benefit: benefit
+      }
+    }.to_json
+  end
+
+  let(:dob) { '2002-12-22' }
+  let(:involvement_type) { "applicant" }
+  let(:opponent) { false }
+  let(:benefit) { 'yes' }
+
+  let(:schema) do
+    {
+      '$id' => "check_financial_eligibility/schemas/applicant",
+      type: :object,
+      required: [:applicant],
+      properties: {
+        applicant: {
+          type: :object,
+          properties: {
+            date_of_birth: {
+              type: :string,
```

Comment:
```
"date": {
  "type": "string",
  "format": "date"
}
```

Reply:
I can't get this to work - it considers anything valid
```
>>>>>>>>>  /Users/stephenrichards/moj/check/spec/json_spike/applicant_schema_spec.rb:39 <<<<<<<<<<
{"$schema"=>"http://json-schema.org/draft-04/schema", "$id"=>"check_financial_eligibility/schemas/applicant", "type"=>"object", "required"=>["applicant"], "properties"=>{"applicant"=>{"type"=>"object", "properties"=>{"date_of_birth"=>{"type"=>"string", "format"=>"date"}, "involvement_type"=>{"enum"=>["applicant"]}, "has_partner_opponent"=>{"type"=>"boolean"}, "receives_qualifying_benefit"=>{"enum"=>["yes", "no"]}}, "required"=>["date_of_birth", "involvement_type", "has_partner_opponent", "receives_qualifying_benefit"]}}}
>>>>>>>>>  /Users/stephenrichards/moj/check/spec/json_spike/applicant_schema_spec.rb:41 <<<<<<<<<<
{"applicant":{"date_of_birth":"3xx0024-1223","involvement_type":"defendant","has_partner_opponent":false,"receives_qualifying_benefit":"yes"}}
[
    [0] "The property '#/applicant/involvement_type' value \"defendant\" did not match one of the following values: applicant in schema 80f83d28-71d8-594f-ade3-747223b5d850"
]```
no error message for the dodgy date